oci/table_oci_ons_subscription.go

```diff
@@ -243,6 +246,81 @@ func getOnsSubscription(ctx context.Context, d *plugin.QueryData, h *plugin.Hydr
 	return response.Subscription, nil
 }
 
+// This hydrate function is to get delivery policy column
+// In case of list call it just return delivery policy without doing any extra api call as list call  contains delivery policy data
+// In case of get call it calls extra list api using topicId as filer to get delivery policy data from the list call
+func getSubscriptionDeliveryPolicy(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Debug("getSubscriptionDeliveryPolicy")
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("oci.getOnsSubscription", "Compartment", compartment, "OCI_REGION", region)
+
+	policy := deliveryPolicy(ctx, h.Item)
+
+	if policy != nil {
+		return policy, nil
+	}
+
+	equalQuals := d.KeyColumnQuals
+
+	// Return nil, if given compartment_id doesn't match
+	if equalQuals["compartment_id"] != nil && compartment != equalQuals["compartment_id"].GetStringValue() {
+		return nil, nil
+	}
+
+	// Create Session
+	session, err := onsNotificationDataPlaneService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	get := h.Item.(ons.Subscription)
+	request := ons.ListSubscriptionsRequest{
+		CompartmentId: types.String(compartment),
+		TopicId:       types.String(*get.TopicId),
+		Limit:         types.Int(50),
+		RequestMetadata: common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(d.Connection),
+		},
+	}
+
+	// Check for additional filter
```

Comment:
Do we want to check for limits? I don't think we need to since we know that the limit doesn't matter here, since we're just getting 1 subscription's delivery policy and we need to check all subscriptions for the topic anyway